scalpel/src/Text/HTML/Scalpel.hs

```diff
@@ -137,8 +138,12 @@ module Text.HTML.Scalpel (
 ,   matches
 -- ** Executing scrapers
 ,   scrape
+,   scrapeT
 ,   scrapeStringLike
+,   scrapeStringLikeT
 ,   URL
+,   downloadURLSpec
+,   downloadURLSpecWithConfig
```

Comment:
Do these need to be exposed?

Reply:
Yes. It allows for separating the scraping from the fetching of the page. `Scraper` still gets the `scrapeURL` and `scrapeURLWithConfig` convenience functions, but `ScraperT` monads don't have an idiomatic equivalent like `scrapeTURL` or anything like that, such a function would have to work on a constraint like `MonadIO m => ScraperT str m a ...`. I think it makes more sense to provide the `downloadURLSpec` functions to be used at the discretion of the user instead of having to write their own. The usecase with a scraper that will also be downloading data is something like:
```
scrapeMyWay :: URL -> ScraperT String IO a -> IO (Maybe a)
scrapeMyWay url s = scrapeT s =<< liftIO (downloadURLSpec url)